io-engine/src/grpc/v0/nexus_grpc.rs

```diff
@@ -25,16 +25,19 @@ use crate::{
 
 /// Map the internal child states into rpc child states (i.e. the states that
 /// the control plane sees)
-impl From<ChildState> for rpc::ChildState {
-    fn from(child: ChildState) -> Self {
-        match child {
+impl From<&NexusChild<'_>> for rpc::ChildState {
+    fn from(child: &NexusChild) -> Self {
+        if child.is_out_of_sync() {
```

Comment:
hmm so when a child is faulted while rebuilding we report degraded when should be faulted?
Might be better to do this if line 37 instead, only when child is open?

Reply:
is_out_of_sync() implies Open. Probably I should change the func name.

Reply:
Yeah I think better to rename then.